### ISSUE TITLE
[Bug] TreeShaking not working properly (src is included in bundle)

### ISSUE THREAD

**Issue by AbdallahAlhaddad** [nuxt-community/vuetify-module#382]
**Modules versions**
```
"nuxt": "^2.14.0"
"@nuxtjs/vuetify": "^1.11.2",
```

**Bug description**
When building for production, `nuxtjs/vuetify` module doesn't properly apply tree-shake as it includes the `src` directory in the final bundle.

**To Reproduce**
- Create the default nuxt/vuetify project (using `npx create-nuxt-app <project-name>` then select vueitfy as a framework).
- Run the command `npm run build --analyze`
- You will see the vuetify module in `app[hash].js` file with approximate  size of `406 kb` as `src` directory is also included.

- **Vuetify size in nuxt app (`406 kb`):**
![nuxt](https://user-images.githubusercontent.com/54389201/92334082-fed8de00-f08a-11ea-8a04-63cbb0ab8e7e.png)

**Expected behavior**
Only the used components is included in the final bundle in the `lib` directory (no `src` directory is included) as in the official `vuetify` module in normal Vue SPA  project.

- **Vuetify size in normal SPA app (`85.64 kb`):** 
![vue](https://user-images.githubusercontent.com/54389201/92334188-08167a80-f08c-11ea-8202-c4598686e30d.png)

**Comment by kamikazechaser** [comment]
Having a similar issue with the latest version.

```bash
WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  app.9e69a9c.css (334 KiB)
```

*config*
- mode: spa
- target: static
- extractCSS: true
- treeShake: true

**Comment by Barbapapazes** [comment]
While I'm trying to find a solution, I found this project and if you analyze it, you will find that when vuetify is added manually, there is no problem with treeshaking !

**Comment by bitbytebit1** [comment]
Any updates on this @Barbapapazes ?

**Comment by Barbapapazes** [comment]
No, try so many things on this project (nuxt/vuetify) and just find that the vuetify-loader-plugin doesn't work as expected here but well outside so I will continue my investigation! 😉

**Comment by dflupu** [comment]
@Barbapapazes fyi the project you linked doesn't have this issue because it's using vuetify v1. v2 was a complete rewrite.

**Comment by aldarund** [comment]
I`l check whats happening here

**Comment by KaelWD** [comment]
This isn't a bug, only sass is being loaded from src. The bundle analyzer will show different results depending on if mini-extract-css-plugin is used.

**Comment by kamikazechaser** [comment]
@KaelWD The `main.sass` is still large, always > 450 Kib in most builds I have seen from other users (#342). I am not familiar with with the internals, but could this be somehow solved? (with say some loader option?).

**Comment by mathe42** [comment]
@kamikazechaser I hacked around this. I wrote a simple parser that reads my css and removes styles I know (but vuetify can't know)  are bit needed. 
YES this is a hack and not the best solution but I removed a LOT of CSS. 

For me I removed all `.v-application--is-rtl`, `[dir=rtl]`, `.theme--dark` . (Some more stuff but that was the biggest CSS Chunk. My Website is in German so no need for rtl and we only have one theme so no theme-dark needed.

But again DO NOT USE THIS as this is not a good way to do this stuff!!

**Comment by kamikazechaser** [comment]
@mathe42 I tried purgeCSS, it does work. But it can severely break a large application. Its also tedious since you have to handpick and add regex to the safelist. Let me see if I can come up with a safe configuration. I have seen some people try this FullHuman/purgecss#67. This looks unsafe. Its better to use `extractCSS: true` and use the CLI on the produced CSS.

**Comment by mathe42** [comment]
Basicly my function look like this:
```js
function handleContents(content) {
  return content
    .split('}')
    .filter((v) => {
      const sel = v.split('{')[0]
      return sel
        .split(',')
        .every(
          (selector) =>
            !selector.includes('.v-application--is-rtl') &&
            !selector.includes('[dir=rtl]') &&
            !selector.includes('.theme--dark')
        )
    })
    .join('}')
}

```

**Comment by t1mwillis** [comment]
You might also consider parsing through all your vue components to find which Vuetify Atomic CSS helpers are used and removed all those that aren't. I wrote a script you'd need to use in conjunction with `purgecss`/`css-byebye` [here](https://gist.github.com/t1mwillis/4b14716258ea73660586115dc0c965d3)

**Comment by trollderius** [comment]
> You might also consider parsing through all your vue components to find which Vuetify Atomic CSS helpers are used and removed all those that aren't. I wrote a script you'd need to use in conjunction with `purgecss`/`css-byebye` [here](https://gist.github.com/t1mwillis/4b14716258ea73660586115dc0c965d3)
Thank you very much sir

**Comment by VladBrok37** [comment]
> Basicly my function look like this:
> 
> ```js
> function handleContents(content) {
>   return content
>     .split('}')
>     .filter((v) => {
>       const sel = v.split('{')[0]
>       return sel
>         .split(',')
>         .every(
>           (selector) =>
>             !selector.includes('.v-application--is-rtl') &&
>             !selector.includes('[dir=rtl]') &&
>             !selector.includes('.theme--dark')
>         )
>     })
>     .join('}')
> }
> ```

@mathe42 could you please help, where did you put this code in your nuxt project?
